src/mem/alloc.h

```diff
@@ -330,15 +339,14 @@ namespace snmalloc
         return;
       }
       dealloc_sized_slow(p_auth, p_free, size);
-#endif
     }
 
     SNMALLOC_SLOW_PATH
     void
     dealloc_sized_slow(AuthPtr<void> p_auth, FreePtr<void> p_free, size_t size)
```

Comment:
Do these register pressure concerns affect the codegen without StrictProvenance? `dealloc_sized_slow` will not be inlined, so the calling of them even in tail calls may affect the codegen for the fast path.